### PR TITLE
A preliminary version of Jugaad

### PR DESCRIPTION
**Pull request by nisargjhaveri**
Needs proper manual review and testing.

Making PR at this stage so that others can also work on this.

### CODE REVIEW

**Comment by hharchani** [comment]
You should update the fork. #40 is completed.

**Comment by nisargjhaveri** [comment]
Updated. It still fails, I'll update it soon.

File: src/static/styles/common.css [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+.notifArea .notif {
+    margin-top: 0.5em;
+    background: #eee;
+    padding: 0.7em;
+}
+
+.notifArea .notif .notifMsg {
+}
```
Comment (hharchani) [inline src/static/styles/common.css:65]: Ye hai :P

Comment (nisargjhaveri) [inline src/static/styles/common.css:65]: Hota hai kabhi kabhi.. Is liye to linters hote hai. :P

**Comment by nisargjhaveri** [comment]
Looks fine to me. But still needs testing. Someone? Anyone? Please?

**Comment by hharchani** [comment]
I think you should work upon increasing scrutinizer ratings :stuck_out_tongue:

**Comment by nisargjhaveri** [comment]
Some of the issues are invalid and many are "out of scope". :P So I think we need to work on configuring it.

**Comment by hharchani** [comment]
No excuses @nisargjhaveri

**Comment by hharchani** [comment]
1. Why is Travis CI build is failing?
2. This is just the preliminary version right? We're looking only for the basic functionalities right? I mean there are some obvious fixes, such that at many places pages are called as file, or if someone doesn't have any permission, they get a blank page as soon as they login, etc.
3. Why `perms_model`, why not `permission_model`?
4. I feel that the jugaad model and jugaad controller are pretty large files. If they can be split up, that would be great :P 

There are some more issues, but I don't think that should be fixed in this commit :stuck_out_tongue: 
Rest all looks fine. If these all are fine too, I give it a +1 :+1:

**Comment by nisargjhaveri** [comment]
1. Because some changes in JShint config, I think. Some of them are stupidity of me, some are of JShint. I'll fix mine. Someone look at JShint :P
2. It should not give blank page if someone does not have permission, if it does so, it is wrong, I'll have to look at it. Can you provide how can I reproduce that error?
3. Because it is short. :P
4. Yes, it would be great it we can split those up, but I don't know how feasible and desirable it is.

About the preliminary version, it does not do everything, but what it does, it should do it nicely. So, any obvious bugs should not be allowed. 

Pages are always called files at Jugaad level, and I don't intend to change that. If someone has a really big problem with it, they can try to change it.

**Comment by hharchani** [comment]
Remaining tasks to be done
- Handle Directory's index page.
- Fix the build, if we merge this failing build.
- Fix the blank page issue.
- Write complete build instructions (e.g. import db)

**Comment by hharchani** [comment]
This is ready to merge, with this failing build.

File: src/app/controllers/hello.php [hunk 2]
```diff
@@ -1,7 +0,0 @@
-<?php
-
-class hello extends Controller {
```
Comment (polybuildr) [inline src/app/controllers/hello.php:3]: We will miss you.

**Comment by polybuildr** [comment]
I haven't been able to review this well, but for organisational reasons, merging. We need to fix the issues ASAP, of course.

**Pull request merged**